lib/languages/js/source-code/source-code.js

```diff
@@ -795,6 +795,7 @@ class SourceCode extends TokenStore {
 	 */
 	getIndexFromLoc(loc) {
 		if (
+			loc === null ||
```

Comment:
Can we add a separate error for this? The error message says what it expects but doesn't mention that `loc` is `null`. Something like, "Unexpected null argument."